app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,17 @@
+module Api
+  module V1
+    class BaseController < ApplicationController
+      before_filter :restrict_access
+
+      private
+
+      def restrict_access
+        unless ApiKey.exists?(access_token: params[:token])
```

Comment:
here I'd add a check for !banned? even if you just stub banned? out to always return false for now -- would be better to ban problem keys and thereby keep their records than delete them, IMO